ClientApp/KachnaOnline/src/app/shared/services/authentication.service.ts

```diff
@@ -340,8 +344,19 @@ export class AuthenticationService {
     this.changeGamificationConsent(this.user.gamificationConsent);
   }
 
-  updateNicknameRequest() {
-    return this.http.put<any>(`${USERS_API}/me/nickname`, JSON.stringify(this.user.nickname),
+  updateNickname(nickname: string) {
```

Comment:
This is a bit tricky. KIS only allows nickname changes with certain frequency (one change per XX days). The way nicknames are handled on the backend is just wrong, I shouldn't have added the PUT /nickname endpoint at all because it doesn't really make sense, the KO nickname should always be in sync with KIS. I'll have to adjust the implementation of this in the backend.

For the time being, I suggest handling this on frontend using this approach:
- You first send a nickname change request to **KIS** (PUT .../kis/api/users/me/nickname).
- If it returns 200, you post the request to KO (and you don't even have to signalise the result of this operation)
- If it returns 409, you signalise error "Požadovanou přezdívku už někdo používá.", for 429 you signalise "Přezdívka může být změněna jen jednou za XX dní."; in both cases, you _don't_ send a request to KO.

Reply:
(I'm not sure what the timeout is, when I find it out, I'll update this comment.)

Reply:
Done. When the precise timeout is known, I will update the error message.

Reply:
Done.